os/src-jvm/package.scala

```diff
@@ -59,6 +59,8 @@ package object os {
 
   val sub: SubPath = SubPath.sub
 
+  val checker: DynamicVariable[Checker] = new DynamicVariable[Checker](Checker.Nop)
```

Comment:
How about adding these for ergonomics?

```scala
def withChecker[T](checker: Checker)(op: => T): T = {
    checker.withValue(checker)(op)
  }
  
def checkReadAccess(path: os.Path): Unit = checker.value.onRead(path)
def checkWriteAccess(path: os.Path): Unit = checker.value.onWrite(path)
```

Reply:
I think for now we can punt on these, since it's a pretty advanced feature so being easy to use isn't as important

Reply:
Whats the benefit here?
The number of characters typed at the call site is roughly the same.